### DISCUSSION TITLE
Access frontmatter via mdx

### DISCUSSION THREAD

**Discussion by devjiwonchoi** [shuding/nextra#2280] [Q&A]
I'm currently importing an mdx file into my mdx in pages.

I want to do something like:

```mdx
// original file

---
title: Introduction
description: Welcome to the Next.js Documentation.
---

Welcome to the Next.js documentation!
```

into:

```mdx
// /pages/index.mdx

import Introduction from 'docs/index.mdx'

# {frontmatter.title}

<Introduction />
```

P.S. When I try to use `useConfig`, it gives me this:

```mdx
import { useConfig } from 'nextra-theme-docs'
import Introduction from 'docs/index.mdx'

export const frontMatter = useConfig().frontMatter

<Introduction />
```

```warning
TypeError: Cannot read properties of null (reading 'useContext')
```

Assuming `useConfig` is only allowed inside `theme.config.tsx`.

**Comment by devjiwonchoi** [comment]
Duplicate of #1508, and my issue was that while embedding mdx on mdx, the frontmatter is left behind, resulting in frontmatter = {}.

Short answer for now:

```tsx
// theme.config.tsx

import { useConfig } from 'nextra-theme-docs'

// ...
main: ({ children }) => {
  const { frontMatter } = useConfig()
  return (
    <>
      <h1 className="nx-mt-2 nx-text-4xl nx-font-bold nx-tracking-tight">
        {frontMatter.title}
      </h1>
      {children}
    </>
  )
},
// ...
```

**Reply by mkschreder** [reply]
Does this even still work in v4? It seems useConfig has been removed from nextra theme docs

**Reply by dimaMachina** [reply]
@mkschreder in v4 it was moved in `normalizePagesResult.activeMetadata` https://nextra.site/docs/docs-theme/api#useconfig-hook
